pkg/server/settingswatcher/settings_watcher.go

```diff
@@ -250,25 +250,46 @@ func (s *SettingsWatcher) handleKV(
 		}
 	}
 
+	s.maybeSet(ctx, name, settingsValue{
+		val:       val,
+		ts:        kv.Value.Timestamp,
+		tombstone: tombstone,
+	})
+	if s.storage != nil {
+		return kv
+	}
+	return nil
+}
+
+// maybeSet will update the stored value and the corresponding setting
+// in response to a kv event, assuming that event is new.
+func (s *SettingsWatcher) maybeSet(ctx context.Context, name string, sv settingsValue) {
 	s.mu.Lock()
 	defer s.mu.Unlock()
+	if existing, ok := s.mu.values[name]; ok && sv.ts.Less(existing.ts) {
+		return
```

Comment:
nit: a comment here (or on the func) as to why it'd be out of order might be nice; I'd ague I'd have expected it here more than on the struct.

Reply:
Done.